pgmpy/tests/test_models/test_NaiveBackdoorRegressor.py

```diff
@@ -0,0 +1,354 @@
+"""
+Tests for Naive Backdoor Regressor.
+"""
+
+import numpy as np
+import pandas as pd
+import pytest
+from sklearn.dummy import DummyRegressor
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.linear_model import LinearRegression
+from sklearn.utils.estimator_checks import parametrize_with_checks
+
+from pgmpy.base import DAG
+from pgmpy.prediction.NaiveBackdoorRegressor import NaiveBackdoorRegressor
+
+
+def make_estimator():
+    """Create a valid estimator for sklearn compatibility tests."""
+    dag = DAG(
+        ebunch=[("Z", "X"), ("Z", "Y"), ("X", "Y")],
+        roles={"exposure": "X", "outcome": "Y", "adjustment": ["Z"]},
+    )
+    return NaiveBackdoorRegressor(causal_graph=dag, base_estimator=LinearRegression())
+
+
+@parametrize_with_checks([make_estimator()])
+def test_sklearn_compatibility(estimator, check):
```

Comment:
Another issue I think is that sklearn checks `X` with arrays of different sizes (including a single column array), but in our case, we need `X` to have at least `number of exposure + number of adjustment + number of pre-treatment` columns. A simple solution for avoiding this could be to use a causal graph that has an empty adjustment set for the estimator checks instead of the one defined in `make_estimator`.